### PR TITLE
bump build wait for slow dancer mysql builds with lots of prereqs to …

### PR DESCRIPTION
**Pull request by gabemontero**
…download

the dancer-mysql are consistently 10 to 15 minutes now with all the prereq downloads and image pushes ... seen about a dozen failures where the just miss the cutoff if you take the 2 minutes for the build to exist and formerly 10 to complete 

image-eco has been pretty flaky wrt .... the perl builds have also been skirting the edge

/assign @bparees

### CODE REVIEW

File: test/extended/util/framework.go [hunk 1]
```diff
@@ -707,7 +707,7 @@ func WaitForABuild(c buildv1clienttyped.BuildInterface, name string, isOK, isFai
 		return err
 	}
 	// wait longer for the build to run to completion
-	err = wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
+	err = wait.Poll(5*time.Second, 20*time.Minute, func() (bool, error) {
```
Comment (bparees) [inline test/extended/util/framework.go]: i'm pretty sure the e2e framework itself is capping out individual tests at 15 mins.  We need to stay under that because when the framework kills the test, we get zero information about the failure.

@smarterclayton can you confirm the ginkgo test timeout?

Comment (gabemontero) [reply test/extended/util/framework.go]: ok it that is true, and assuming the 15 min cap isn't changing, I may start getting inclined to just remove some of these big/fat/slow sample combos from image eco

Comment (bparees) [reply test/extended/util/framework.go]: See if you can run it down w/ SteveK or Clayton (confirming that there's a cap and what it is)

Also can you look at some of the successful runs and see how long the test took in those?  I'm wondering if we're always right on the edge, or if it's just that we occasionally hiccup.

Comment (gabemontero) [reply test/extended/util/framework.go]: the tests don't print the build details if they succeed ... of if that data is somewhere I've never known where to find it .... any ideas?

Comment (gabemontero) [reply test/extended/util/framework.go]: yeah looked at successful run https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-samples-operator/103/pull-ci-openshift-cluster-samples-operator-master-e2e-aws-image-ecosystem/237/artifacts/e2e-aws-image-ecosystem/ and I can't find the build pods from the tests

couldn't find the e2e test pod either, but even if I just missed it, as I noted before, we don't dump all those diagnostics on success

Comment (bparees) [reply test/extended/util/framework.go]:
```
started: (0/34/42) "[image_ecosystem][Slow] openshift sample application repositories [image_ecosystem][perl] test perl images with dancer-ex db repo  Building dancer-mysql app from new-app should build a dancer-mysql image and run it in a pod [Suite:openshift]"

passed: (20.8s) 2019-02-13T02:27:35 "[image_ecosystem][Slow] openshift images should be SCL enabled  returning s2i usage when running the image \"centos/perl-520-centos7\" should print the usage [Suite:openshift]"
```

and

```
started: (0/41/42) "[image_ecosystem][Slow] openshift images should be SCL enabled  using 
the SCL in s2i images \"centos/python-36-centos7\" should be SCL enabled [Suite:openshift]
"

passed: (4m42s) 2019-02-13T02:30:30 "[image_ecosystem][Slow] openshift sample application repositories [image_ecosystem][php] test php images with cakephp-ex db repo  Building cakephp-mysql app from new-app should build a cakephp-mysql image and run it in a pod [Suite:openshift]"
```

as seen in:
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/22029/pull-ci-openshift-origin-master-e2e-aws-image-ecosystem/131/artifacts/e2e-aws-image-ecosystem/container-logs/test.log.gz

Comment (gabemontero) [reply test/extended/util/framework.go]: oh the entire test time was sufficient for you :-)

Comment (bparees) [reply test/extended/util/framework.go]: (no details about where time is spent, but it tells us we're not normally running up against the limit, assuming those are the right tests i picked out, which is all i wanted to get at)

Comment (gabemontero) [reply test/extended/util/framework.go]: we just can't tell how long those dancer prereq downloads took when it was faster ... when it fails, the build completes, but when you look at the build logs, it is spending 8 or 9 minutes dowloading all that stuff

Comment (bparees) [reply test/extended/util/framework.go]: Right.  Again my initial interest was "are we normally running 9 mins and if it takes even slightly longer we fail" vs "we're normally running 4 mins and we only fail when something goes unusually wrong".

seems like we're in the latter case, which means this may get better again depending on the external factors causing the issue.

Comment (gabemontero) [reply test/extended/util/framework.go]: yeah *maybe* ... I've been seeing these intermittent delays coming up on 2 weeks

I'll circle back to the framework cap and we can go from there
